docs/Developers/Git_Basics.rst

```diff
@@ -374,30 +374,112 @@ branch in your fork. If it says your branch is ``n commits behind
+    # r, reword <commit> = use commit, but edit the commit message
+    # e, edit <commit> = use commit, but stop for amending
+    # s, squash <commit> = use commit, but meld into previous commit
+    # f, fixup <commit> = like "squash", but discard this commit's log message
+    # x, exec <command> = run command (the rest of the line) using shell
+    # b, break = stop here (continue rebase later with 'git rebase --continue')
+    # d, drop <commit> = remove commit
+    # l, label <label> = label current HEAD with a name
+    # t, reset <label> = reset HEAD to a label
+    # m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
+    # .       create a merge commit using the original merge commit's
+    # .       message (or the oneline, if no original merge commit was
+    # .       specified). Use -c <commit> to reword the commit message.
+    #
+    # These lines can be re-ordered; they are executed from top to bottom.
+    #
+    # If you remove a line here THAT COMMIT WILL BE LOST.
+    #
+    # However, if you remove everything, the rebase will be aborted.
+    #
+    # Note that empty commits are commented out
+
+Git offers a quick view of several commands that we can apply to each commit. 
+For our purposes we will focus on five.
+
+Picking
+-------
+
+As simple as it sounds. if you want to use a commit without changing it, use pick.
+By default, interactive rebase will pick all your commits.
+
+Rewording
+---------
+
+Equivalent to the `git commit --amend` except it applies to the specified commit. Use
+reword if your commit message does not follow proper formatting. Depending on the number 
+of rewords applied, new editors will appear for each commit message to amend. Make your changes
+and exit the editor to apply them. 
+
+Editing
+-------
+
+Similar to pick, except you can make changes to the commit before it is applied. Make any
+changes and run 
+
+:: 
+    
+    $ git commit --amend
+
+To apply the patched changes to the commit before it's applied to the branch.
+
 .. note::
 
-    master is the SHA1 hash of the commit before which you want to squash all
-    the commits and make sure that rebase is done onto master branch.
+    Be careful of merge conficts resulting from editing older commits. The older the commit,
+    the more likely it is to cause a merge conflict. If your rebase ends up in a bad state, run
 
-An editor will be fired up with all the commits in your current branch
-(ignoring merge commits), which come after the given commit. Keep the first one
+    ``$ git rebase --abort``
+    
+    This will restore the original state of your branch before the rebase was initiated.
+
+Squashing
+---------
+
+It's possible that you have more than one commit and you want them to be
+squashed into a single commit. You can take your series of commits and squash
+them down into a single commit with the interactive rebasing tool. To squash
+your commits run the following command:
+
+Keep the first one
```

Comment:
This line is a bit too small, we can restructure this paragraph a bit to not waste any line space.